lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1052,6 +1052,7 @@ static bool releaseCapturedArgsOfDeadPartialApply(PartialApplyInst *PAI,
   // point.
   for (auto *FinalRelease : Tracker.getFinalReleases()) {
     Builder.setInsertionPoint(FinalRelease);
+    Builder.setCurrentDebugScope(FinalRelease->getDebugScope());
```

Comment:
@vedantk: This is a good example for the upcoming SILBuilder ergonomics work.

Reply:
I've been thinking about this too. FWIW, pretty much every case I found of broken scope recently was the builder updating the insertion point and not updating the current scope.

IMHO there are two opportunities here:
1) Create a fresh builder here instead of updating the insertion point. It should be really cheap to do so.
2) Change the scope as we're changing the insertion point. This is really risky because we don't have a lot of coverage for that yet (but we're improving!) so we need to audit and evaluate whether there are legitimate cases where you want to update the insertion point but not the scope (I can't think of anything off the top of my head, but chances are these cases exist).